.github/workflows/semgrep.yml

```diff
@@ -16,7 +16,7 @@ jobs:
   analyze:
     name: Scan
     runs-on: ubuntu-latest
-    if: (github.actor != 'dependabot[bot]' || github.actor != 'renovate[bot]')
+    if: ${{github.actor != 'dependabot[bot]' && github.actor != 'renovate[bot]' && (!github.event.pull_request || github.event.pull_request.head.repo.full_name == github.repository)}}
```

Comment:
`!github.event.pull_request ||`
does not that mean it never runs when a PR is submitted?

Reply:
https://math.stackexchange.com/questions/2178839/why-is-a-only-if-b-equivalent-to-not-a-or-b

Here I was checking if A was not defined, if it's not defined I'm in not in a PR, so I should run the check, otherwise, if it's a PR I should run only if it's a PR starting from an internal branch.